cdap-ui/app/cdap/components/Reports/ReportsDetail/RunsPagination/index.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {handleRunsPageChange} from 'components/Reports/store/ActionCreator';
+import PaginationWithTitle from 'components/PaginationWithTitle';
+
+function RunsPaginationView({totalCount, totalPages, currentPage}) {
+  return (
+    <PaginationWithTitle
+      handlePageChange={handleRunsPageChange}
+      currentPage={currentPage}
+      totalPages={totalPages}
+      title={totalCount > 1 ? "Runs" : "Run"}
+      numberOfEntities={totalCount}
+    />
+  );
+}
+
+RunsPaginationView.propTypes = {
+  totalCount: PropTypes.number,
+  totalPages: PropTypes.number,
+  currentPage: PropTypes.number
+};
+
+const mapStateToProps = (state) => {
+  let {
+    totalRunsCount: totalCount,
+    runsOffset,
+    runsLimit
+  } = state.details;
+
+  let totalPages = Math.ceil(totalCount / runsLimit);
+  let currentPage;
+
+  if (runsOffset === 0) {
+    currentPage = 1;
+  } else {
+    currentPage = Math.ceil((runsOffset + 1) / runsLimit);
+  }
```

Comment:
this logic should be inside the `RunsPaginateView`. The `mapStateToProps` should only direct map from store value to the props for the component.